### PR TITLE
Integrate auth token gem into project.

### PR DESCRIPTION
**Pull request by eschlange**
* Added auth token config.
* Initialized devise model.
* Added participant model that ties to tokens.

[#96195532]

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Devise model which allows access to dashboard.
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
```
Comment (ericcf) [inline app/models/user.rb]: Make sure you turn off `registerable`.

File: app/models/participant.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# A participant in the SiS study.
+class Participant < ActiveRecord::Base
+  # Available smoking statuses
+  SMOKING_STATUS_PRE_CESSATION = "pre_cessation"
+  SMOKING_STATUS_CESSATION = "cessation"
+  SMOKING_STATUS_POST_CESSATION = "post_cessation"
+
+  has_one :token_auth_authentication_token,
```
Comment (ericcf) [inline app/models/participant.rb]: You probably need to specify `foreign_key: "entity_id"` here.

File: app/models/participant.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+          dependent: :destroy,
+          foreign_key: :entity_id
+  has_one :token_auth_configuration_token,
```
Comment (ericcf) [inline app/models/participant.rb]: Same here :-)

**Comment by eschlange** [comment]
Requested updates are complete: :checkered_flag:

**Pull request merged**